### ISSUE TITLE
Support `query_as` on `ActiveRel` instances

### ISSUE THREAD

**Issue by cheerfulstoic** [neo4jrb/neo4j#1081]
Maybe on the class level as well?  Don't know if that makes sense because we're generally avoiding querying `ActiveRel` classes

**Comment by cheerfulstoic** [comment]
See: http://stackoverflow.com/questions/34429496/query-as-using-neo4j-5-2-3-giving-nil-class

**Comment by subvertallchris** [comment]
I don't think so, all ActiveRel query methods except `find` (which is really just an alias for `Neo4j::Relationship.load`) will be removed completely in the next version.

**Comment by cheerfulstoic** [comment]
But wouldn't a `query_as` instance method be of the same ilk as `find`?  You'd be identifying a particular relationship and providing a variable for it at the start of a `Query` chain

**Comment by subvertallchris** [comment]
Can't they just do `rel_instance.to_node.query_as(:n)`? Encouraging queries to start from rels isn't good practice, we already have a really powerful set of tools for querying from nodes and labels.

**Comment by cheerfulstoic** [comment]
I'm not sure what `rel_instance` is and I don't think I'm seeing it.  But would you call `to_node` on a relationship instance?  I'm confused...

I'm thinking of `ActiveRel#query_as` as starting the query off with something like:

```cypher
MATCH (from:FromLabel)-[rel_var:REL_TYPE]->(to:ToLabel)
WHERE from.uuid = {from_uuid} AND to.uuid = {to_uuid}
```

**Comment by subvertallchris** [comment]
`rel_instance` would be an instance of an ActiveRel class. You'd call `to_node` on it if you want to start a match from that related node, although you'd then have to match back over the rel or `match_to` the other node if you want to get back to where you started...

Still, I don't see what problem this is solving? In that SO post, he can already do this:

```ruby
User.agendas(:a, :r).rel_where(action_type: Action::TYPE[:stance_change], action_allowed: true).pluck(:r)
```

**Comment by subvertallchris** [comment]
Really, my big issue with it is that it seems like a slippery slope leading to the expectation of a reimplementation of more query methods, which is exactly what we had and what we're trying to get away from. If you think it'd be really helpful, by all means, go for it, but I'd just like to be sure that it isn't better solved by advising users to approach it differently.
